monkey/monkey_island/cc/services/agent_plugin_service/mongo_agent_plugin_repository.py

```diff
@@ -0,0 +1,245 @@
+import logging
+from enum import Enum
+from typing import Any, Dict, Optional
+
+import gridfs
+from pymongo import MongoClient
+
+from common import OperatingSystem
+from common.agent_plugins import AgentPlugin, AgentPluginManifest, AgentPluginType
+from monkey_island.cc.repositories import (
+    RemovalError,
+    RetrievalError,
+    StorageError,
+    UnknownRecordError,
+)
+
+from .i_agent_plugin_repository import IAgentPluginRepository
+
+BINARY_OS_MAPPING_KEY = "binaries"
+
+logger = logging.getLogger(__name__)
+
+
+class MongoAgentPluginRepository(IAgentPluginRepository):
+    def __init__(self, mongo_client: MongoClient) -> None:
+        self._agent_plugins_collection = mongo_client.monkey_island.agent_plugins
+        self._agent_plugins_binaries_collections = self._get_binary_collections(mongo_client)
+
+    def _get_binary_collections(
+        self, mongo_client: MongoClient
+    ) -> Dict[OperatingSystem, gridfs.GridFS]:
+        # Note: This expects the database to have a collection for each operating system
```

Comment:
Do they get automatically created if they don't exist?

Reply:
They are because of the call to `gridfs.GridFS`